packages/shared/src/governance.ts

```diff
@@ -0,0 +1,897 @@
+  await relayClient.dev.setStorage({
+    System: {
+      account: [
+        [[defaultAccounts.bob.address], { providers: 1, data: { free: 10e10 } }],
+        [[defaultAccounts.charlie.address], { providers: 1, data: { free: 10e10 } }],
+        [[defaultAccounts.dave.address], { providers: 1, data: { free: 10e10 } }],
+        [[defaultAccounts.eve.address], { providers: 1, data: { free: 10e10 } }],
+      ],
+    },
+  })
+
+  /**
+   * Get current referendum count i.e. the next referendum's index
+   */
+  const referendumIndex = await relayClient.api.query.referenda.referendumCount()
+
+  /**
+   * Submit a new referendum
+   */
+
+  const submitReferendumTx = relayClient.api.tx.referenda.submit(
+    {
+      Origins: 'SmallTipper',
+    } as any,
+    {
+      Inline: relayClient.api.tx.treasury.spendLocal(1e10, defaultAccounts.bob.address).method.toHex(),
+    },
+    {
+      After: 1,
+    },
+  )
+  const submitReferendumEvents = await sendTransaction(submitReferendumTx.signAsync(defaultAccounts.alice))
+
+  await relayClient.dev.newBlock()
+
+  // Fields to be removed, check comment below.
+  let unwantedFields = new RegExp('index')
+  await checkEvents(submitReferendumEvents, 'referenda')
+    .redact({ removeKeys: unwantedFields })
+    .toMatchSnapshot('referendum submission events')
+
+  /**
+   * Check the created referendum's data
+   */
+
+  let referendumDataOpt: Option<PalletReferendaReferendumInfoConvictionVotingTally> =
+    await relayClient.api.query.referenda.referendumInfoFor(referendumIndex)
+  assert(referendumDataOpt, "submitted referendum's data cannot be `None`")
+  let referendumData: PalletReferendaReferendumInfoConvictionVotingTally = referendumDataOpt.unwrap()
+  // These fields must be excised from the queried referendum data before being put in the test
+  // snapshot.
+  // These fields contain epoch-sensitive data, which will cause spurious test failures
+  // periodically.
+  unwantedFields = new RegExp('alarm|submitted')
+  await check(referendumData)
+    .redact({ removeKeys: unwantedFields })
+    .toMatchSnapshot('referendum info before decision deposit')
+
+  assert(referendumData.isOngoing)
+  // Ongoing referendum data, prior to the decision deposit.
+  const ongoingRefPreDecDep: PalletReferendaReferendumStatusConvictionVotingTally = referendumData.asOngoing
+
+  assert(ongoingRefPreDecDep.alarm.isSome)
+  const undecidingTimeoutAlarm = ongoingRefPreDecDep.alarm.unwrap()[0]
+  const blocksUntilAlarm = undecidingTimeoutAlarm.sub(ongoingRefPreDecDep.submitted)
+  // Check that the referendum's alarm is set to ring after the (globally predetermined) timeout
+  // of 14 days, or 201600 blocks.
+  assert(blocksUntilAlarm.eq(relayClient.api.consts.referenda.undecidingTimeout))
+
+  // The referendum was above set to be enacted 1 block after its passing.
+  assert(ongoingRefPreDecDep.enactment.isAfter)
+  await check(ongoingRefPreDecDep.enactment.asAfter).toMatchObject(1)
```

Comment:
for simple checks that you don't need redact or other features implemented by check, you can use `expect` from `vitesst`
```suggestion
  expect(ongoingRefPreDecDep.enactment.asAfter).toBe(1)
```